src/Common/src/TypeSystem/IL/Stubs/DelegateMarshallingMethodThunk.cs

```diff
@@ -68,8 +70,7 @@ public override MethodSignature Signature
             {
                 if (_signature == null)
                 {
-                    // TODO: Parse UnmanagedFunctionPointerAttribute 
-                    bool isAnsi = true;
+                    bool isAnsi = PInvokeMetadata.GetCharSet(_delegateType.GetDelegatePInvokeAttributes()) == PInvokeAttributes.CharSetAnsi;
```

Comment:
I can see that you are using PInvokeMetadata.GetCharSet to decode it (and perhaps encode the defaults). One interesting way to make this more discoverable is to change GetDelegatePinvokeAttribute to return struct of a single field, and add helpers on it.